src/libsync/creds/oauth.cpp

```diff
@@ -497,16 +497,9 @@ QUrl OAuth::authorisationLink() const
     const QUrl url = _authEndpoint.isValid()
         ? Utility::concatUrlPath(_authEndpoint, {}, query)
         : Utility::concatUrlPath(_serverUrl, QStringLiteral("/index.php/apps/oauth2/authorize"), query);
-    return url;
-}
 
-void OAuth::authorisationLinkAsync(std::function<void (const QUrl &)> callback) const
-{
-    if (_wellKnownFinished) {
-        callback(authorisationLink());
-    } else {
-        connect(this, &OAuth::authorisationLinkChanged, callback);
-    }
+    Q_ASSERT(!url.isEmpty());
```

Comment:
How would that be possible?

Reply:
We rather need to check for _wellKnownFinished ?

Reply:
I did want to have some assertion in there at least. We can increase the checks' granularity of course.